### ISSUE TITLE
redis-store options not working as documented

### ISSUE THREAD

**Issue by ratneshraval** [redis-store/redis-store#315]
I'm trying to create my custom store and utilize Redis-store as the main workhorse.

First I tried inhering from `Redis::Rails` and `Redis::Store`. But none of them worked. 
So I ended up inhering from recommended `ActionDispatch::Session::AbstractStore` (Noted that Redis-rails or redis-store are NOT inherited from `AbstractStore`.

Repo with test branch: https://github.com/ratneshraval/cwds_store/tree/redis-options

Questions:
1. What's the best way to make a custom store that is a really lightweight wrapper on redis-store?
2. Initializing with options in this way works
https://github.com/ratneshraval/cwds_store/blob/e477d34d40e782d55f5dd21117b02c18d9d4a690/lib/cwds_store/store.rb#L14-L21
```
     # works: host, port, namespace
     # doesn't work: expire_after
      @redis = Redis::Store.new(
        {
          host: options[:host],
          port: options[:port],
          db: 0,
          namespace: 'test-session',
          expire_after: 3.minutes
        }
      )
```
But initializing with options shown in documentation does NOT work 
https://github.com/ratneshraval/cwds_store/blob/e477d34d40e782d55f5dd21117b02c18d9d4a690/lib/cwds_store/store.rb#L25-L36
```
      # doesn't work at all
      @redis = Redis::Store.new(
        {
          servers: [
            {
              host: options[:host],
              port: options[:port],
              db: 0,
              namespace: 'test-session'
            }
          ],
          expire_after: 3.minutes
        }
      )
```
3. Why expiration is not set even though initialized with correct option? 

When initializing as session_store in rails application, the documented options approach DOES WORK. So not sure why [this](https://github.com/ratneshraval/cwds_store/blob/e477d34d40e782d55f5dd21117b02c18d9d4a690/lib/cwds_store/store.rb#L25-L36) does not work

**Comment by tubbo** [comment]
The [ActionDispatch::Session::RedisStore](https://github.com/redis-store/redis-actionpack/blob/master/lib/action_dispatch/middleware/session/redis_store.rb) class takes a slightly different options syntax than everything else, taking the array of server arguments with `:servers`. You can see the `#initialize` method is pulling the `:servers` argument out and setting it as `:redis_server`.

Not sure what's up with `:expire_after`, that should be working here...

**Comment by tubbo** [comment]
Looked a little more into the issue just now. The reason why the `:expire_*` settings aren't taking effect when you pass them into the constructor is because they are explicitly passed by whatever class is using the `Redis::Store` upon each request. They're used in [Redis::Store::Ttl](https://github.com/redis-store/redis-store/blob/master/lib/redis/store/ttl.rb), and this whole feature is admittedly a bit confusing.

What kind of store are you building on top of `Redis::Store`? There might be a more suitable base class to work with (like `Redis::Rack` for Rack session stores without ActionDispatch).

**Comment by ratneshraval** [comment]
@tubbo Thank you for looking into this quickly.

I was able to get what I needed with `ActionDispatch::Session::RedisStore`. So thank you very much.
I'm surprised I didn't run into that with my research before I started down the path of `AbstractStore`.

The master branch is updated with what I was trying to do. Just override and provide simpler and controlled options to all micro services utilizing my gem, keeping them similar.
